### PR TITLE
Feature/shoot controls

### PR DESCRIPTION
**Pull request by Jazy5552**
Implemented shooting controls by use of the InputController and the ShootController.

### CODE REVIEW

**Review by AleksandrHovhannisyan — CHANGES_REQUESTED** [review]
Not sure I like the momentum, it's kinda weird if you move backwards and shoot because then the projectiles are actually slower. What do you think?

Other than that, looks good :) Be sure to resolve the conflict.

File: Assets/Scripts/Entity/Player/InputController.cs [hunk 1]
```diff
@@ -2,19 +2,31 @@
 using System.Collections.Generic;
 using UnityEngine;
 
+[RequireComponent(typeof(Player))]
 [RequireComponent(typeof(PlayerMovement))]
+[RequireComponent(typeof(ShootController))]
 public class InputController : MonoBehaviour
 {
+    private Player p;
```
Comment (AleksandrHovhannisyan) [inline Assets/Scripts/Entity/Player/InputController.cs]: I would use more descriptive names. I know it's faster to develop if you use shorter variable names, but `p` isn't very informative :D

File: Assets/Scripts/Entity/Player/Player.cs [hunk 2]
```diff
@@ -19,7 +20,7 @@ public class Player : Entity
     void Awake()
     {
         SetEnabled(true);
-        SetWeapon(weapon);
+        //SetWeapon(weapon);
```
Comment (AleksandrHovhannisyan) [inline Assets/Scripts/Entity/Player/Player.cs:28]: Why comment this out? Did you forget to uncomment after debugging or? :)

File: ProjectSettings/InputManager.asset [hunk 3]
```diff
@@ -37,6 +37,102 @@ InputManager:
...
+    axis: 0
+    joyNum: 0
+  - serializedVersion: 3
+    m_Name: FireVertical
+    descriptiveName: 
+    descriptiveNegativeName: 
+    negativeButton: joystick button 0
+    positiveButton: joystick button 3
+    altNegativeButton: joystick button 16
+    altPositiveButton: joystick button 19
```
Comment (AleksandrHovhannisyan) [inline ProjectSettings/InputManager.asset:111]: What are buttons 16 and 19? They don't appear in the chart that Cameran linked a while ago.

File: ProjectSettings/InputManager.asset [hunk 4]
```diff
@@ -37,6 +37,102 @@ InputManager:
...
+    type: 0
+    axis: 0
+    joyNum: 0
+  - serializedVersion: 3
+    m_Name: FireHorizontal
+    descriptiveName: 
+    descriptiveNegativeName: 
+    negativeButton: joystick button 2
+    positiveButton: joystick button 1
+    altNegativeButton: joystick button 18
```
Comment (AleksandrHovhannisyan) [inline ProjectSettings/InputManager.asset:62]: Same here, Unity doesn't have any joystick button > 9.

File: Assets/Scripts/Entity/Player/InputController.cs (hunk 1 above)
Comment (Jazy5552) [reply Assets/Scripts/Entity/Player/InputController.cs]: Fair enough.

File: Assets/Scripts/Entity/Player/Player.cs (hunk 2 above)
Comment (Jazy5552) [reply Assets/Scripts/Entity/Player/Player.cs:28]: The SetWeapon, and all weapon logic, should now live in the ShootController. I simply commented it out to reduce any conflicts and allow for a refactor later.

File: ProjectSettings/InputManager.asset (hunk 3 above)
Comment (Jazy5552) [reply ProjectSettings/InputManager.asset:111]: They are the same xbox controller buttons, but on MAC. http://wiki.unity3d.com/index.php/Xbox360Controller

File: ProjectSettings/InputManager.asset (hunk 4 above)
Comment (Jazy5552) [reply ProjectSettings/InputManager.asset:62]: For MAC, http://wiki.unity3d.com/index.php/Xbox360Controller

**Comment by Jazy5552** [comment]
Yeha, I'm not a fan of the momentum at all. Just figured I would implement a simple checkbox so we could check it out. The whole, moving slower while player moves backwards, could easily be resolved by projecting the velocity vector alone the right vector of the rigidbody. I did fix it, just so you can see how to feels.

File: Assets/Scripts/Entity/Player/Player.cs (hunk 2 above)
Comment (AleksandrHovhannisyan) [reply Assets/Scripts/Entity/Player/Player.cs:28]: Ahh, right right

File: ProjectSettings/InputManager.asset (hunk 3 above)
Comment (AleksandrHovhannisyan) [reply ProjectSettings/InputManager.asset:111]: Gotcha, so we're supporting both Mac and Windows basically? Or does this only support Mac atm?

**Review by AleksandrHovhannisyan — DISMISSED** [review]

**Comment by Jazy5552** [comment]
Merging due to previous review.

**Pull request merged**